mmdet/models/backbones/xception.py

```diff
@@ -0,0 +1,317 @@
+        super(Block, self).__init__()
+        self.batch_norm = False if norm_cfg is None else len(norm_cfg) != 0
+        self.out_channels = out_filters
+        if out_filters != in_filters or stride != 1:
+            self.side = build_conv_layer(
+                conv_cfg,
+                in_filters,
+                out_filters,
+                1,
+                stride=stride,
+                bias=False)
+            if self.batch_norm:
+                _, self.bn = build_norm_layer(norm_cfg, out_filters)
+        else:
+            self.side = None
+
+        self.relu = nn.ReLU(inplace=True)
+        rep = []
+
+        filters = in_filters
+        if grow_first:
+            rep.append(self.relu)
+            rep.append(
+                SeparableConv(
+                    conv_cfg,
+                    in_filters,
+                    out_filters,
+                    3,
+                    stride=1,
+                    padding=1,
+                    bias=False))
+            if self.batch_norm:
+                rep.append(build_norm_layer(norm_cfg, out_filters))
+            filters = out_filters
+
+        for i in range(reps - 1):
+            rep.append(self.relu)
+            rep.append(
+                SeparableConv(
+                    conv_cfg,
+                    filters,
+                    filters,
+                    3,
+                    stride=1,
+                    padding=1,
+                    bias=False))
+            if self.batch_norm:
+                rep.append(build_norm_layer(norm_cfg, filters))
+
+        if not grow_first:
+            rep.append(self.relu)
+            rep.append(
+                SeparableConv(
+                    conv_cfg,
+                    in_filters,
+                    out_filters,
+                    3,
+                    stride=1,
+                    padding=1,
+                    bias=False))
+            if self.batch_norm:
+                rep.append(build_norm_layer(norm_cfg, out_filters))
+
+        if not start_with_relu:
+            rep = rep[1:]
+        else:
+            rep[0] = nn.ReLU(inplace=False)
+
+        if stride != 1:
+            if conv_cfg['type'] == 'Conv3d':
+                rep.append(nn.MaxPool3d(3, stride, 1))
+            elif conv_cfg['type'] == 'Conv1d':
+                rep.append(nn.MaxPool1d(3, stride, 1))
+            else:
+                rep.append(nn.MaxPool2d(3, stride, 1))
+        self.rep = nn.Sequential(*rep)
+
+    def forward(self, inp):
+        x = self.rep(inp)
+
+        if self.skip is not None:
+            skip = self.side(inp)
+            if self.batch_norm:
+                skip = self.bn(skip)
+        else:
+            skip = inp
+
+        x += skip
+        return x
+
+
+@BACKBONES.register_module()
+class Xception(nn.Module):
+    """Xception optimized for the ImageNet dataset, as specified in
+    https://arxiv.org/pdf/1610.02357.pdf."""
+
+    def __init__(self,
```

Comment:
1. Please add complete docstrings for all newly added modules and functions as those in MMDetection V2.x.